media/server/main/include/MediaPipelineServerInternal.h

```diff
@@ -397,22 +399,35 @@ class MediaPipelineServerInternal : public IMediaPipelineServerInternal, public
     void scheduleNotifyNeedMediaData(MediaSourceType mediaSourceType);
 
     /**
-     * @brief Set volume internally, only to be called on the main thread.
+     * @brief Set the target volume level with a transition internally, only to be called on the main thread.
+     *        Sets the volume with a transition effect over the specified duration and easing type.
      *
-     * @param[in] volume Target volume level (0.0 - 1.0)
+     * @param[in] targetVolume : Target volume level (0.0 - 1.0)
+     * @param[in] duration : Duration of the volume transition in milliseconds
+     * @param[in] type : Easing type for the volume transition
      *
-     * @retval true on success false otherwise
+     * @retval true on success, false otherwise
      */
-    bool setVolumeInternal(double volume);
+    bool setVolumeInternal(double targetVolume, uint32_t duration = 0, EaseType type = EaseType::EASE_LINEAR);
```

Comment:
You dont need to include defaults in here, only need it in the interface (one place)

Reply:
Or remove this method as we dont need internal methods yet.